org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.java

```diff
@@ -159,7 +161,10 @@ public class LanguageServerImpl implements LanguageServer, WorkspaceService, Tex
 	private static final Logger LOG = Logger.getLogger(LanguageServerImpl.class);
 
 	@Inject
-	private RequestManager requestManager;
+	private RequestManager persistedStateRequestManager;
```

Comment:
Please note: There must be exactly one request manager. It is basically the synchronization primitive as used by the Xtext LSP server. Having two breaks the entire approach taken towards read / write access

Reply:
The read/write semantics implemented by the request manager really only makes sense for a single index. Once you introduce a second index, then you do not want reads and writes to the first to be blocked by writes to the second (for example), but the desired behavior for requests to one of the indexes is the same as the existing request manager. There are other ways the requests could be managed (one manager with multiple queues for instance), but the simplest approach was just to have two request managers. I am happy to refactor if you think this should be handled differently.